flow/connectors/utils/aws.go

```diff
@@ -20,79 +24,158 @@ type AWSSecrets struct {
-	if awsRegion == "" {
-		return nil, errors.New("AWS_REGION must be set")
-	}
+type AWSCredentials struct {
+	AWS         aws.Credentials
+	EndpointUrl *string
+}
 
-	awsEndpoint := creds.Endpoint
-	if awsEndpoint == "" {
-		awsEndpoint = os.Getenv("AWS_ENDPOINT")
-	}
+type AWSCredentialsProvider interface {
+	Retrieve(ctx context.Context) (AWSCredentials, error)
+	GetUnderlyingProvider() aws.CredentialsProvider
+	GetRegion() string
+}
 
-	awsKey := creds.AccessKeyID
-	if awsKey == "" {
-		awsKey = os.Getenv("AWS_ACCESS_KEY_ID")
+type ConfigBasedAWSCredentialsProvider struct {
+	config aws.Config
+}
+
+func (r *ConfigBasedAWSCredentialsProvider) GetUnderlyingProvider() aws.CredentialsProvider {
+	return r.config.Credentials
+}
+
+func (r *ConfigBasedAWSCredentialsProvider) GetRegion() string {
+	return r.config.Region
+}
+
+// Retrieve should be called as late as possible in order to have credentials with latest expiry
+func (r *ConfigBasedAWSCredentialsProvider) Retrieve(ctx context.Context) (AWSCredentials, error) {
+	retrieved, err := r.config.Credentials.Retrieve(ctx)
+	if err != nil {
+		return AWSCredentials{}, err
 	}
+	return AWSCredentials{
+		AWS:         retrieved,
+		EndpointUrl: r.config.BaseEndpoint,
+	}, nil
+}
+
+func NewConfigBasedAWSCredentialsProvider(config aws.Config) AWSCredentialsProvider {
+	return &ConfigBasedAWSCredentialsProvider{config: config}
+}
+
+type StaticAWSCredentialsProvider struct {
+	credentials AWSCredentials
+	region      string
+}
+
+func (s *StaticAWSCredentialsProvider) GetUnderlyingProvider() aws.CredentialsProvider {
+	return credentials.NewStaticCredentialsProvider(s.credentials.AWS.AccessKeyID, s.credentials.AWS.SecretAccessKey,
+		s.credentials.AWS.SessionToken)
+}
+
+func (s *StaticAWSCredentialsProvider) GetRegion() string {
+	return s.region
+}
 
-	awsSecret := creds.SecretAccessKey
-	if awsSecret == "" {
-		awsSecret = os.Getenv("AWS_SECRET_ACCESS_KEY")
+func (s *StaticAWSCredentialsProvider) Retrieve(ctx context.Context) (AWSCredentials, error) {
+	return s.credentials, nil
+}
+
+func NewStaticAWSCredentialsProvider(credentials AWSCredentials, region string) AWSCredentialsProvider {
+	return &StaticAWSCredentialsProvider{
+		credentials: credentials,
+		region:      region,
 	}
+}
 
-	awsRoleArn := creds.AwsRoleArn
-	if awsRoleArn == "" {
-		awsRoleArn = os.Getenv("AWS_ROLE_ARN")
+func getPeerDBAWSEnv(connectorName string, awsKey string) string {
+	return os.Getenv(fmt.Sprintf("PEERDB_%s_AWS_CREDENTIALS_%s", strings.ToUpper(connectorName), awsKey))
+}
+
+func LoadPeerDBAWSEnvConfigProvider(connectorName string) AWSCredentialsProvider {
+	accessKeyId := getPeerDBAWSEnv(connectorName, "AWS_ACCESS_KEY_ID")
+	secretAccessKey := getPeerDBAWSEnv(connectorName, "AWS_SECRET_ACCESS_KEY")
+	region := getPeerDBAWSEnv(connectorName, "AWS_REGION")
+	endpointUrl := getPeerDBAWSEnv(connectorName, "AWS_ENDPOINT_URL")
+	var endpointUrlPtr *string
+	if endpointUrl != "" {
+		endpointUrlPtr = &endpointUrl
 	}
```

Comment:
```suggestion
	var endpointUrlPtr *string
	if endpointUrl != "" {
		endpointUrlPtr = &endpointUrl
	}
```

At end you can use `aws.String(endpointUrl)` instead of having `endpointUrlPtr` variable

https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.51.3/aws#String